src/verify_expr.ml

```diff
@@ -1841,30 +1858,33 @@ module VerifyExpr(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
       let rec iter chunks vs h fields =
         match fields with
           [] ->
-          let (_, csym, _, _) = List.assoc sn struct_accessor_map in
-          cont chunks h (Some (if consumeUninitChunk then real_unit (* dummy term; should never be used *) else ctxt#mk_app csym (List.rev vs)))
+          let (_, csym, _, _, csym_opt) = List.assoc sn struct_accessor_map in
+          let csym_symb = if consumeUninitChunk then csym_opt else csym in
+          let struct_val = ctxt#mk_app csym_symb (List.rev vs) in
+          cont chunks h (Some struct_val)
         | (f, (lf, gh, t0, offset, finit))::fields ->
           let t = instantiate_type tpenv t0 in
           match t with
             StaticArrayType (_, _) | StructType _ | UnionType _ ->
             consume_c_object_core_core l coefpat (field_address l typeid_env addr sn targs f) t h typeid_env true consumeUninitChunk $. fun chunks' h (Some value) ->
-            let value =
-              if consumeUninitChunk then value else prover_convert_term value t t0
-            in
+            let value = prover_convert_term value t t0 in

```

Comment:
If `consumeUninitChunk` is true, `value` is not of type `t`, right? So in that case it's not okay to convert from t to t0?